packages/docs-infra/src/pipeline/loaderUtils/pathToFileUrl.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Converts a file system path to a file:// URL in an isomorphic way.
+ *
+ * This handles the difference between Unix and Windows paths:
+ * - Unix: `/home/user/file.ts` → `file:///home/user/file.ts`
+ * - Windows: `C:/Users/file.ts` → `file:///C:/Users/file.ts`
+ *
+ * The key insight is that Windows absolute paths start with a drive letter (e.g., C:/)
+ * and need an extra slash to form a valid file:// URL.
+ *
+ * @param filePath - The file system path to convert (should use forward slashes)
+ * @returns The file:// URL
+ */
+export function pathToFileUrl(filePath: string): string {
```

Comment:
the [built-in](https://nodejs.org/api/url.html#urlpathtofileurlpath-options) doesn't work?